### DISCUSSION TITLE
Configure Kiali to use custom dashboards in Grafana

### DISCUSSION THREAD

**Discussion by genfemme** [kiali/kiali#7319] [Q&A]
We don't have default Istio Dashboards in the Grafana instead some custom dashboards with complex queries setup in the variables section for each dashboard. Can custom dashboards be added to the CR?
In the doc - https://kiali.io/docs/configuration/p8s-jaeger-grafana/grafana/ there is mention of default istio dashboards variables. Are they the only ones that can be used? We have chain, application, pod as variables. The variable "namespace" is used in most of the dashboard variables that we have.

**Comment by jmazzitelli** [comment]
I don't know anything about the Grafana dashboard integration. I'll let someone else chime in who knows. But I do know there is some other documentation (not much) that is also found in the Kiali CR reference that might help: 

https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.grafana.dashboards[*].name

**Reply by genfemme** [reply]
Thanks @jmazzitelli 
it only shows the options for the default Istio Dashboards.

**Reply by genfemme** [reply]
@jmazzitelli 
Is there anyone else who can help with this please?

**Reply by jshaughn** [reply]
https://kiali.io/docs/configuration/custom-dashboard/ ?

**Reply by nrfox** [reply]
It looks like the variables you can set are (for whatever reason) predefined: https://github.com/kiali/kiali/blob/master/config/dashboards/dashboards_config.go#L81-L87. My guess is this feature assumes you are using the predefined istio dashboards. You can open up a feature request to allow for custom grafana dashboards and variables to be set.

**Reply by jmazzitelli** [reply]
Do not confuse the "custom dashboards" with the Grafana dashboards. They are different things. The code @nrfox pointed to and the docs @jshaughn pointed to are for the custom dashboards (these are not related to the Grafana dashboards). The Grafana dashboards are configured via this block of config: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.external_services.grafana.dashboards[*]

They are documented here: https://kiali.io/docs/configuration/p8s-jaeger-grafana/grafana/

And please be aware, as that documentation states:

```
Although Kiali offers similar views in its metrics dashboards, it is not in Kiali’s goals
to provide the advanced querying options, nor the highly customizable settings,
that are available in Grafana. Thus, it is recommended that you use
Grafana if you need those advanced options.
```

**Reply by nrfox** [reply]
@jmazzitelli what I linked is used by `GetGrafanaInfo`: https://github.com/kiali/kiali/blob/087e0834056f574f65517e8379703d7468808196/business/grafana.go#L25-L63 what is what the `/api/grafana` route returns. I haven't dug into the frontend part of it but I'm assuming that is what is used by the frontend to generate links to the **grafana** dashboards and not the custom kiali dashboards.

**Reply by nrfox** [reply]
afaict what's being asked for is the ability to provide links to custom grafana dashboards with variables other than the ones that come with predefined istio dashboards and afaict that's not supported in kiali with the grafana integration today hence why it'd be a feature request.

**Reply by jmazzitelli** [reply]
OK, so I guess the Grafana dashboards piggy-back on the Kiali custom dashboards framework (told you I don't know much about the grafana dashboards ;)

But as our docs indicate, we try to avoid getting in the business of implementing advanced querying options/customizable settings for Grafana... if you can't get the basics to show in Kiali, it is best to use Grafana directly.
